packages/dashboard-frontend/src/services/registry/devfiles.ts

```diff
@@ -29,7 +30,35 @@ function resolveIconUrl(metadata: che.DevfileMetaData, baseUrl: string): string
   return createURL(metadata.icon, baseUrl).href;
 }
 
-export function resolveLinks(metadata: che.DevfileMetaData, baseUrl: string): any {
+export function resolveTags(
+  metadata: che.DevfileMetaData,
+  baseUrl: string,
+  isExternal: boolean,
+): string[] {
+  const resolvedTags: string[] = [...metadata.tags];
+  if (isExternal) {
+    const resourceTag = new URL(baseUrl).host
+      .replace(/^registry\./, '')
+      .replace(/^\w/, char => char.toUpperCase());
+    resolvedTags.push(resourceTag);
+  }
+  return resolvedTags;
+}
+
+export function resolveLinks(
+  metadata: che.DevfileMetaData,
+  baseUrl: string,
+  isExternal: boolean,
+): any {
+  if (isExternal) {
+    const self = metadata.links?.self;
+    if (self && !metadata.links.v2) {
+      const devfileCatalog = new URL('devfiles', baseUrl).toString();
+      const v2 = self.replace(':', '/').replace('devfile-catalog', devfileCatalog);
+      metadata.links.v2 = v2;
+      delete metadata.links.self;
+    }
```

Comment:
Do I get it correctly that this code only works with links from devfile.io? Will it work with another external registry?

Reply:
I mean that another external registry (not the `registry.devfile.io`) may have a different folder structure so this line
```typescript
const v2 = self.replace(':', '/').replace('devfile-catalog', devfileCatalog);
```
will be irrelevant and cause runtime exceptions.

Reply:
Ok. I have modified this code:
```
      const v2 = self.startsWith('devfile-catalog')
        ? self.replace(':', '/').replace('devfile-catalog', devfileCatalog)
        : self;
```